Rules/AvoidGlobalFunctions.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+using System;
+using System.Collections.Generic;
+#if !CORECLR
+using System.ComponentModel.Composition;
+#endif
+using System.Globalization;
+using System.Management.Automation.Language;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+#if !CORECLR
+    [Export(typeof(IScriptRule))]
+#endif
+    public class AvoidGlobalFunctions : AstVisitor, IScriptRule
+    {
+        private List<DiagnosticRecord> records;
+        private string fileName;
+
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null)
+            {
+                throw new ArgumentNullException(Strings.NullAstErrorMessage);
+            }
+
+            records = new List<DiagnosticRecord>();
+            this.fileName = fileName;
+
+            ast.Visit(this);
+
+            return records;
+        }
+
+        #region VisitCommand functions
+        public override AstVisitAction VisitFunctionDefinition(FunctionDefinitionAst functionDefinitionAst)
+        {
+            if (functionDefinitionAst.Name.StartsWith("Global:", StringComparison.OrdinalIgnoreCase) && IsModule())
```

Comment:
Since IsModule checks for filename extension, you can move this check one level up to AnalyzeScript method.